event/event.go

```diff
@@ -0,0 +1,181 @@
+// event is a package that models events that occur during the execution of a program
+package event
+
+// Event is an interface that custom events should implement.
+// It is empty for now because there are no general methods
+// required of events yet.
+type Event interface {
+}
+
+// KeyCharacter is an event that occurs when a character is actually typed on
+// the keyboard. This may be provided by an input method.
+type KeyCharacter struct {
+	// Code of the key pressed or released
+	Code int
+	// Key board modifiers
+	Modifiers int
+	// Character typed
+	Character rune
+}
+
+// KeyDown is an event that occurs when a key is pressed on the keyboard.
+type KeyDown struct {
+	// Code of the key pressed or released
+	Code int
+	// Key board modifiers
+	Modifiers int
+}
+
+// KeyUp is an event that occurs when a key is released on the keyboard.
+// The data is the same as for a KeyDown event
+type KeyUp KeyDown
+
+// GamepadAxis is for event where an axis on a game pad changes.
+type GamepadAxis struct {
+	// Which gamepad caused the event
+	ID int
+	// Axis of the game pad that changed, if any
+	Axis int
+	// Position of the axis after the change
+	Position float32
+}
+
+//GamepadButtonDown is a game pad button press event.
+type GamepadButtonDown struct {
+	// Which gamepad caused the event
+	ID int
+	// Button that was pressed, if any.
+	Button int
+	// Position of the button after the change.
+	Position float32
```

Comment:
What is position?

Reply:
On some game pads, buttons are analog, and can report how "hard" they are pressed. Maybe the word Pressure is better? Pressure normally varies between 0.0 (not pressed) and 1.0 (fully pressed).

Reply:
Pressure sounds better :)